packages/http-server-csharp/src/lib/service.ts

```diff
@@ -226,7 +226,24 @@ export async function $onEmit(context: EmitContext<CSharpServiceEmitterOptions>)
         return code`${csType.type.getTypeReference(this.emitter.getContext()?.scope)}`;
       }
 
-      return code`${this.emitter.emitTypeReference(elementType, this.emitter.getContext())}[]`;
+      return this.collectionDeclaration(elementType, array);
+    }
+
+    collectionDeclaration(elementType: Type, array: Model): EmitterOutput<string> {
+      const isUniqueItems = getUniqueItems(this.emitter.getProgram(), array);
+      const collectionType = isUniqueItems
+        ? CollectionType.ISet
+        : CSharpServiceOptions.getInstance().collectionType;
+
+      switch (collectionType) {
+        case CollectionType.ISet:
+          return code`ISet<${this.emitter.emitTypeReference(elementType, this.emitter.getContext())}>`;
```

Comment:
I feel like we should have one instance of IEnumerable<IEnumerable<T>> in the tests.  We can probably reuse the test you added with different config settings for collection type.

Reply:
added :)